mava/components/jax/building/adders.py

```diff
@@ -43,6 +46,18 @@ def name() -> str:
         """Static method that returns component name."""
         return "executor_adder"
 
+    @staticmethod
+    def required_components() -> List[Type[Callback]]:
+        """List of other Components required in the system for this Component to function.
+
+        BaseTrainerInit required to set up builder.store.table_network_config.
+        BaseSystemInit required to set up builder.store.unique_net_keys.
+
+        Returns:
+            List of required component classes.
+        """
+        return [BaseTrainerInit, BaseSystemInit]
```

Comment:
The diagram is a little confusing with both abstract classes and subclasses are shown. I have been trying to see why the circular dependency exists but it does not make sense that Step has so many dependencies when it is an abstract class with no implementations. Do you agree?

Reply:
Discussed this: abstract classes do depend on other abstract classes but only as required